letsencrypt/client.py

```diff
@@ -135,14 +140,38 @@ def obtain_certificate(self, domains, csr=None):
                 M2Crypto.X509.load_request_der_string(csr.data)),
             authzr)
 
-        # Save Certificate
-        cert_path, chain_path = self.save_certificate(
-            certr, self.config.cert_path, self.config.chain_path)
-
-        revoker.Revoker.store_cert_key(
-            cert_path, self.account.key.file, self.config)
-
-        return cert_key, cert_path, chain_path
+        cert_pem = certr.body.as_pem()
+        chain_pem = None
+        if certr.cert_chain_uri:
+            chain_pem = self.network.fetch_chain(certr)
+
+        if chain_pem is None:
+            # XXX: just to stop RenewableCert from complaining; this is
+            #      probably not a good solution
+            chain_pem = ""
+        else:
+            chain_pem = chain_pem.as_pem()
+
+        return cert_pem, cert_key.pem, chain_pem
+
+    def obtain_and_enroll_certificate(self, domains, authenticator, installer,
+                                      csr=None):
+        """Get a new certificate for the specified domains using the specified
+        authenticator and installer, and then create a new renewable lineage
+        containing it."""
+        cert_pem, privkey, chain_pem = self._obtain_certificate(domains, csr)
+        # TODO: Add IPlugin.name or use PluginsFactory.find_init instead
+        #       of assuming that each plugin has a .name attribute
+        self.config.namespace.authenticator = authenticator.name
```

Comment:
Perhaps we should put the names of the plugins into the <tt>IConfig</tt> interface.  But in any case it seems we can kick the search for the most elegant way to do this to a future ticket.